src/util/formatJS.js

```diff
@@ -0,0 +1,26 @@
+const format = require("prettier").format;
+const redent = require("redent");
+
+module.exports = function(code, printContext, indent) {
+  const config = {
+    semi: !printContext.noSemi,
+    printWidth: printContext.maxLen,
+    singleQuote: printContext.singleQuote,
+    useTabs: printContext.indentString[0] === '\t',
+    tabWidth: printContext.indentString.length
+  };
+
+  if (code.slice(0, 5) === 'class') {
+    // When parsing classes prettier (babylon) requires a class name.
+    // Marko does not, we get around this by parsing as an expression.
+    code = format('(' + code + ');', config).slice(1, -3);
```

Comment:
Should this be `.slice(1, -2)` instead? Where is the extra character coming from?

Reply:
It adds a newline as well in this case. I updated the PR to make this more clear by instead using `trim`.

Reply:
Thanks for clarifying and adding `trim`.